File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.LinkedList;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        LinkedList<String> cache = new LinkedList<>();
        
        for(String city : cities) {
            city = city.toLowerCase();
            
            if(cacheSize > 0) {
                if(!cache.contains(city)) {
                    if(cache.size() == cacheSize) {
                        cache.removeFirst();
                    }
                    cache.add(city);
                    answer += 5;
                }
                else {
                    cache.remove(city);
                    cache.add(city);
                    answer += 1;
                }
            }
            else {
                answer += 5;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    cache = []
    
    for city in cities:
        city = city.lower()
        if cacheSize > 0:
            if city not in cache:
                if len(cache) == cacheSize:
                    cache.pop(0)
                cache.append(city)
                
                answer += 5
            else:
                cache.pop(cache.index(city))
                cache.append(city)
                
                answer += 1
        else:
            answer += 5
            
    return answer